src/cloudai/workloads/nccl_test/prediction_report_generator.py

```diff
@@ -0,0 +1,250 @@
+# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+# Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+import subprocess
+from pathlib import Path
+from typing import Tuple
+
+import pandas as pd
+
+from cloudai import TestDefinition
+from cloudai.report_generator.tool.csv_report_tool import CSVReportTool
+
+
+class NcclTestPredictionReportGenerator:
+    """Generate NCCL test predictor reports by extracting and analyzing performance data."""
+
+    def __init__(self, collective_type: str, output_path: Path, test_definition: TestDefinition):
+        self.collective_type = collective_type
+        self.output_path = output_path
+        self.stdout_path = output_path / "stdout.txt"
+        self.test_definition = test_definition
+        self.predictor = test_definition.predictor
+
+        if not self.predictor:
```

Comment:
Let's not warn twice. I would even suggest this message should be "debug" only as part of `generate()`. Content should say that "predictor is not enabled", by this time "install" part should already be completed successfully and such message will be very confusing.